### PR TITLE
Team page

### PR DESCRIPTION
**Pull request by Nafisay**

### CODE REVIEW

File: src/components/TeamTab.astro [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      </a>
+    ))
+  }
+</div>
+
+<style lang="scss">
+  @import "../styles/index.scss";
+
+  .navbar {
+    // position: fixed;
```
Comment (harshkhandeparkar) [inline src/components/TeamTab.astro:45]: Please remove these lines if they are no longer required.

File: src/components/TeamTab.astro [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    width: 2rem;
+    height: 45rem;
+    z-index: 100;
+    //border-radius: 12px;
+
+    text-align: left;
+    //writing-mode: vertical-lr;
+    text-orientation: mixed;
+
+    //@include raised;
```
Comment (harshkhandeparkar) [inline src/components/TeamTab.astro:67]: Here too.

File: src/components/TeamTab.astro [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    }
+  }
+
+  span {
+    width: 6cqi;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+  .navbar-logo {
```
Comment (harshkhandeparkar) [inline src/components/TeamTab.astro:145]: Rename these classes too, remove `navbar`.

File: src/components/TeamTab.astro [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+  }
+  .navbar-logo span {
+    width: 7.7cqi;
+    height: 7.7cqi;
+    justify-content: flex-start;
+  }
+</style>
+
+<script>
+  let toTop = document.querySelector('[href="#"]');
```
Comment (harshkhandeparkar) [inline src/components/TeamTab.astro:166]: I believe this is no longer required.

File: src/pages/team/[team].astro [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import TeamCard from "../../components/TeamCard.astro";
+import TeamTab from "../../components/TeamTab.astro";
+
+import ADVISORS from "../../data/advisors.json";
+import CTMS from "../../data/ctms.json";
+import EXECUTIVES from "../../data/execs.json";
+
+export function getStaticPaths() {
+  return [
+    { params: { team: "Core Team Members" } },
```
Comment (harshkhandeparkar) [inline src/pages/team/[team].astro:14]:
```suggestion
    { params: { team: "ctm" } },
```
Rename the links here too.

File: src/pages/team/[team].astro [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    { params: { team: "Executives" } },
```
Comment (harshkhandeparkar) [inline src/pages/team/[team].astro:15]:
```suggestion
    { params: { team: "executives" } },
```

File: src/pages/team/[team].astro [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    { params: { team: "Advisors" } },
```
Comment (harshkhandeparkar) [inline src/pages/team/[team].astro:16]:
```suggestion
    { params: { team: "advisors" } },
```

File: src/pages/team/[team].astro [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  ];
+}
+
+const team = Astro.params.team as string;
+
+var team_members = ADVISORS;
+if (team === "Core Team Members") {
```
Comment (harshkhandeparkar) [inline src/pages/team/[team].astro:23]:
```suggestion
if (team === "ctm") {
```

File: src/pages/team/[team].astro [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+  return [
+    { params: { team: "Core Team Members" } },
+    { params: { team: "Executives" } },
+    { params: { team: "Advisors" } },
+  ];
+}
+
+const team = Astro.params.team as string;
+
+var team_members = ADVISORS;
```
Comment (harshkhandeparkar) [inline src/pages/team/[team].astro:22]:
```suggestion
let team_members = ADVISORS;
```

`var` is deprecated. See [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let) for more information.

File: src/pages/team/[team].astro [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+if (team === "Core Team Members") {
+  team_members = CTMS;
+}
+if (team === "Executives") {
```
Comment (harshkhandeparkar) [inline src/pages/team/[team].astro:26]:
```suggestion
if (team === "executives") {
```

File: src/pages/team/[team].astro [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+---
+
+<Layout title="Meet our Team">
+  <main>
+    <div class="container">
+      <TeamTab />
+      <ul role="list" class="link-card-grid">
+        {
+          team_members.length > 0 &&
+            team_members.map((advisor) => (
```
Comment (harshkhandeparkar) [inline src/pages/team/[team].astro:38]:
```suggestion
            team_members.map((member) => (
```
:)

File: src/pages/team/[team].astro [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+  </main>
+</Layout>
+<style>
+  main {
+    padding: 2rem;
+    max-width: 1000px;
+    width: 100%;
+    margin: 0 auto;
+    margin-bottom: 10%;
+    /*max-width: calc(100% - 2rem);
```
Comment (harshkhandeparkar) [inline src/pages/team/[team].astro:53]: Remove these comments too, if they are not required.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: src/components/TeamCard.astro [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+    flex-direction: column;
+    align-items: center;
+    justify-content: top;
+    text-align: center;
+    align-content: space-around;
+    max-width: 100%;
+    width: 13rem;
+    max-height: 100%;
+    height: 16rem;
+    z-index: -1;
```
Comment (harshkhandeparkar) [inline src/components/TeamCard.astro:40]:
```suggestion
```

File: src/pages/team/[team].astro [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+import CTMS from "../../data/ctms.json";
+import EXECUTIVES from "../../data/execs.json";
+
+export function getStaticPaths() {
+  return [{ params: { team: "ctms" } }, { params: { team: "executives" } }, { params: { team: "advisors" } }];
+}
+
+const team = Astro.params.team as string;
+
+let team_members = ADVISORS;
```
Comment (harshkhandeparkar) [inline src/pages/team/[team].astro:16]: It would be better to use a switch...case statement here.

File: src/components/TeamCard.astro [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+  linkedin_url: string;
+  github_username: string;
+}
+
+const { name, linkedin_url, github_username } = Astro.props as Props;
+---
+
+<li class="link-card">
+  <div>
+    <img src={`https://avatars.githubusercontent.com/${github_username}`} alt="PFP" class="pfp" />
```
Comment (harshkhandeparkar) [inline src/components/TeamCard.astro:14]: If the `alt` property is set, let's set it to something useful like `Profile Picture` or `${name}'s Profile Picture`

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: .yarn/releases/yarn-1.22.19.cjs
Comment (rajivharlalka) [inline .yarn/releases/yarn-1.22.19.cjs:1]: Is this needed?

**Review by rajivharlalka — CHANGES_REQUESTED** [review]
Moreover switching between CTM,EXec and Adv chages the entire page and goes back to top. Can this be corrected? @chirag-ghosh

File: .yarn/releases/yarn-1.22.19.cjs
Comment (chirag-ghosh) [reply .yarn/releases/yarn-1.22.19.cjs:1]: For some reason, Nafisa ke system mai chahiye. I actually want to look into it and see what's the issue

**Comment by rajivharlalka** [comment]
> I have now set it to automatically scroll down when we change the team. This however is not a good solution. Ideally, the component should update there itself. I think instead of using URLs, we can have a react component (to take advantage of reactive states) to achieve this. But let's keep it for sometime later.

@harshkhandeparkar  can you create an issue for this so that it doesnt get left out, Atleast we'll have soemthing working right now

**Review by harshkhandeparkar — COMMENTED** [review]
Should we make a separate tab for alumni?

**Comment by chirag-ghosh** [comment]
You mean like the other 3? @rajivharlalka  and I thought to keep the alums section always present in the page. WHat do you think will be better?

**Comment by harshkhandeparkar** [comment]
> You mean like the other 3? @rajivharlalka and I thought to keep the alums section always present in the page. WHat do you think will be better?

Since there are many alums, that section will be too long imo.

**Review by chirag-ghosh — APPROVED** [review]

**Review by rajivharlalka — APPROVED** [review]
LGTM since chirag reviewed

**Review by harshkhandeparkar — APPROVED** [review]
Who am I to suggest changes to Chirag

**Comment by chirag-ghosh** [comment]
And there goes the flat heirarchy. Anyways....

**Pull request merged**